src/ngx_http_datadog_module.cpp

```diff
@@ -498,6 +592,13 @@ static bool is_server_block_begin(const ngx_conf_t *conf) {
          str(*static_cast<const ngx_str_t *>(conf->args->elts)) == "server";
 }
 
+static ngx_str_t block_type(const ngx_conf_t *conf) {
+  if (conf->args == nullptr) {
+    return ngx_string("");
+  }
+  return to_ngx_str(conf->pool, str(*static_cast<const ngx_str_t *>(conf->args->elts)));
```

Comment:
May I suggest to define an `enum` with all the different type of block? So, the set of value is defined and checking against a specific value will be easier and less error-prone. 

````cpp
// rn
if (str(loc_conf->block_type) == "http") {
   ...
}

// with enum
if (loc_conf->block_type == BlockType::HTTP) {
   ...
}
````

Reply:
I considered that, but then stayed with the string for two reasons:

1. The string names of the relevant blocks (server, http, location) are never going to change.
2. There could be, in theory, future values that I don't anticipate. Why enumerate "these three and then 'other'," when I could just use the strings directly.

There's another argument, too, albeit weaker. When I see some code that reifies the notion of, say, `BlockType`, I take that to mean that the code is expressing a constraint of the design. In our case, though, we don't control the set of block types, and so to enumerate the values only serves to give the values names. But the values _are_ names.